### PR TITLE
vscode-runner: init at 1.6.1

### PR DESCRIPTION
**Pull request by pinage404**
## Description of changes

KRunner plugin for quickly opening recent VSCode workspaces 
https://github.com/Merrit/vscode-runner

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
	(i don't know how to fully restart KRunner)
   need to reboot to be really correctly integrated in KRunner
   tested on Plasma 6 Wayland
   don't know if it still works with Plasma 5 or X11
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by pinage404** [comment]
maybe it need a review from the @qt-kde team, i don't know how this works, i ping some random member @SuperSandro2000

File: pkgs/by-name/vs/vscode-runner/package.nix [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+  kdePackages,
+}:
+
+let
+  pname = "vscode-runner";
+  version = "1.6.1";
+
+  src = fetchFromGitHub {
+    owner = "Merrit";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/vs/vscode-runner/package.nix:15]:
```suggestion
    repo = "vscode-runner";
```
see pname tracking issue

File: pkgs/by-name/vs/vscode-runner/package.nix [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+  pubspecLock = lib.importJSON ./pubspec.lock.json;
+
+  dartEntryPoints = {
+    "bin/vscode_runner" = "bin/vscode_runner.dart";
+  };
+
+  preFixup = ''
+    wrapProgram $out/bin/vscode_runner \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath [ sqlite ]}
```
Comment (SuperSandro2000) [inline pkgs/by-name/vs/vscode-runner/package.nix:33]: Maybe we could use patchelf instead

File: pkgs/by-name/vs/vscode-runner/package.nix [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  '';
+
+  postInstall = ''
+    substituteInPlace ./package/codes.merritt.vscode_runner.service \
+      --replace "Exec=" "Exec=$out/bin/vscode_runner"
```
Comment (SuperSandro2000) [inline pkgs/by-name/vs/vscode-runner/package.nix:38]:
```suggestion
      --replace-fail "Exec=" "Exec=$out/bin/vscode_runner"
```

**Review by SuperSandro2000 — COMMENTED** [review]
otherwise LGTM, didn't try

File: pkgs/by-name/vs/vscode-runner/package.nix (hunk 2 above)
Comment (pinage404) [reply pkgs/by-name/vs/vscode-runner/package.nix:33]: i don't know how to use it

do you have any doc or example with a similar case to help me ?

Comment (SuperSandro2000) [reply pkgs/by-name/vs/vscode-runner/package.nix:33]: Something like https://github.com/NixOS/nixpkgs/blob/7c92ddf469d6d6c236afdcb801a962e9ce03884c/pkgs/applications/misc/holochain-launcher/default.nix#L46 probably

Comment (pinage404) [reply pkgs/by-name/vs/vscode-runner/package.nix]: it seems that it is not needed anymore, it should be legacy from first tests

**Pull request merged**